src/utils/transaction.ts

```diff
@@ -204,8 +203,20 @@ const transaction = {
       });
     }
 
-    if (tx.version === NANO_CONTRACTS_VERSION || tx.version === ON_CHAIN_BLUEPRINTS_VERSION) {
-      const { pubkey } = tx as NanoContract | OnChainBlueprint;
+    let pubkey: Buffer | null = null;
+
+    if (tx.isNanoContract()) {
+      // Get pubkey from nano header
+      const nanoHeader = tx.getNanoHeader();
+      ({ pubkey } = nanoHeader);
+    }
+
+    if (tx.version === ON_CHAIN_BLUEPRINTS_VERSION) {
+      // Get pubkey from ocb tx
+      ({ pubkey } = tx as OnChainBlueprint);
```

Comment:
Object deestructuring a class works but I think we should prefer property access.
```suggestion
      pubkey = (tx as OnChainBlueprint).pubkey;
```

Reply:
It's done in another commit with similar changes: https://github.com/HathorNetwork/hathor-wallet-lib/pull/842/commits/1e0f72f780560ece48ab99251f159585e16674f8